zanata-common-api/src/main/java/org/zanata/common/GlossarySortField.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2015, Red Hat, Inc. and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.zanata.common;
+
+import java.io.Serializable;
+
+/**
+ * @author Alex Eng <a href="aeng@redhat.com">aeng@redhat.com</a>
+ */
+public enum GlossarySortField implements Serializable {
+    src_content("term.content", true),
+    part_of_speech("term.glossaryEntry.pos", true),
+    desc("term.glossaryEntry.description", true),
+    trans_count("term.glossaryEntry.glossaryTerms.size", true);
+
+    private final String entityField;
+    private boolean ascending;
```

Comment:
Instead of having this mutable field into the enum instances, I think you should turn this into a real class, and `getByField()` should return new instances of the class.  (And then you could also make this into a final field.)